### PR TITLE
setup postfix hostname using conf file

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

setup postfix hostname using conf file. REALHOSTNAME can be removed, since HOSTNAME is already set using `hostname -f` or UYUNI_FQDN. After that we can configure postfix using:

`postconf -e myhostname=$HOSTNAME`

## GUI diff

No difference.

- [ ] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- API documentation added: please review the Wiki page [Writing Documentation for the API](https://github.com/uyuni-project/uyuni/wiki/Writing-documentation-for-the-API) if you have any changes to API documentation.
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

related #https://github.com/uyuni-project/uyuni-tools/pull/22

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"    
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mbussolotto** [comment]
not tested yet

**Comment by mbussolotto** [comment]
@cbosdo I'm waiting for a fix in the acceptance tests, but in the meanwhile you can have a look :)

**Comment by mbussolotto** [comment]
it seems it breaks unit tests, having a look on it

File: susemanager/bin/mgr-setup [hunk 1]
```diff
@@ -172,25 +172,10 @@ fi
...
-                done
-        done
-fi
-if [ -n "$REALHOSTNAME" ]; then
-        echo "$REALHOSTNAME" > /etc/hostname
-fi
-# bsc#979664 - SUSE Manager requires a working mail system
-systemctl --quiet enable postfix 2>&1
-systemctl restart postfix
+    postconf -e myhostname=$HOSTNAME
```
Comment (cbosdo) [inline susemanager/bin/mgr-setup:175]: We surely need to persist `/etc/postfix` in a volume with such a change.

**Review by cbosdo — CHANGES_REQUESTED** [review]
Postfix config needs to be persisted in the containers

**Comment by mbussolotto** [comment]
@cbosdo it looks like `susemanger_unittests` are broken for `server-container` branch. I cherry-picked the commit in master and they worked fine. I verified the issue with this PR https://github.com/uyuni-project/uyuni/pull/7447

**Comment by mbussolotto** [comment]
changes for `uyuni-tools` here https://github.com/uyuni-project/uyuni-tools/pull/22/

File: containers/doc/server-kubernetes/pvs.yaml [hunk 2]
```diff
@@ -529,3 +529,28 @@ spec:
...
+    path: /var/uyuni/etc-postfix
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+        - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - uyuni-dev
+
```
Comment (cbosdo) [inline containers/doc/server-kubernetes/pvs.yaml:556]: No longer needed, we can drop the whole file now that it's wired in the helm chart

File: containers/doc/server-kubernetes/migration-job.yaml [hunk 3]
```diff
@@ -39,6 +39,7 @@ spec:
                           /etc/salt \
                           /etc/tomcat \
                           /etc/cobbler \
+                          /etc/postfix \
```
Comment (cbosdo) [inline containers/doc/server-kubernetes/migration-job.yaml:42]: No longer needed now that we are moving the feature to `uyuniadm`. We can drop the whole file.

**Review by cbosdo — COMMENTED** [review]
Please drop the unneeded files or at least the changes in them before merging.

File: containers/doc/server-kubernetes/pvs.yaml (hunk 2 above)
Comment (mbussolotto) [reply containers/doc/server-kubernetes/pvs.yaml]: File removed

File: containers/doc/server-kubernetes/migration-job.yaml (hunk 3 above)
Comment (mbussolotto) [reply containers/doc/server-kubernetes/migration-job.yaml]: File removed

**Review by cbosdo — APPROVED** [review]
LGTM now

**Pull request merged**